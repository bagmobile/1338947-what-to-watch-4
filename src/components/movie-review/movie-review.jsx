import React from "react";
import {movieReviewShape} from "../movie/movie-shape.js";
import moment from "moment";
import {ANONYMOUS_AUTHOR} from "../../settings";

const MovieReview = ({review}) => {
  const {quote, author = ANONYMOUS_AUTHOR, dateTime, rating = 0} = review;

  return (
    <div className="review">
      <blockquote className="review__quote">
        <p className="review__text">{quote}</p>

        <footer className="review__details">
          <cite className="review__author">{author}</cite>
          <time className="review__date"
            dateTime={moment(dateTime).format(`YYYY-MM-DD`)}>
            {moment(dateTime).format(`LL`)}
          </time>
        </footer>
      </blockquote>

      <div className="review__rating">{rating}</div>
    </div>
  );
};

MovieReview.propTypes = {
  review: movieReviewShape
};

export default MovieReview;
