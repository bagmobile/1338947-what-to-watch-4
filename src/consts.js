
export const DEFAULT_GENRE = `All genres`;
export const MAX_GENRES_LIST_SIZE = 10;
export const DEFAULT_MOVIE_LIST_SIZE = 8;
export const MOVIE_LIST_BY_GENRE_SIZE = 4;

export const EMPTY_FIELD_VALUE = `Данные отсутствуют`;
export const ANONYMOUS_AUTHOR = `Анонимный пользователь`;

export const Tab = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const ErrorMessage = {
  INVALID_EMAIL: `Invalid email`,
  INVALID_PASSWORD: `Invalid password`,
  INVALID_USER_DATA: `Invalid user data`,
};

export const LinkPath = {
  INDEX: `/`,
  VIEW_MOVIE: `/movies/:id`,
  LOGIN: `/login`,
  FAVORITE_MOVIES: `/favorite-movies`,
  PLAYER: `/player/:id`
};

export const APIPath = {
  MOVIES: `/films`,
  REVIEWS: `/comments`,
  FAVORITE: `/favorite`,
  PROMO: `/films/promo`,
  LOGIN: `/login`,
};
