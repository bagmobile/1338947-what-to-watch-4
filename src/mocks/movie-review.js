export const mockReviews = [
  {
    id: 1,
    user: {
      id: 1,
      name: `Kate Muir`
    },
    dateTime: `2020-03-07T14:34:25.500Z`,
    quote: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious
              Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in
              years.`,
    rating: 8.8
  },
  {
    id: 2,
    user: {
      id: 2,
      name: `Bill Goodykoontz`
    },
    dateTime: `2020-03-07T14:34:25.500Z`,
    quote: `Anderson's films are too precious for some, but for those of us willing to lose
              ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added
              a hint of gravitas to the mix, improving the recipe.`,
    rating: 8.9
  },
  {
    id: 3,
    user: {
      id: 3,
      name: `Matthew Lickona`
    },
    dateTime: `2020-03-07T14:34:25.500Z`,
    quote: `The mannered, madcap proceedings are often delightful, occasionally silly, and
              here and there, gruesome and/or heartbreaking.`,
    rating: 3.9
  },
  {
    id: 4,
    user: {
      id: 4,
      name: `Amanda Greever`
    },
    dateTime: `2020-03-07T14:34:25.500Z`,
    quote: `I didn't find it amusing, and while I can appreciate the creativity, it's an
              hour and 40 minutes I wish I could take back.`,
    rating: 5
  },
  {
    id: 5,
    user: {
      id: 5,
      name: `Paula Fleri-Soler`
    },
    dateTime: `2020-03-07T14:34:25.500Z`,
    quote: `It is certainly a magical and childlike way of storytelling, even if the content
              is a little more adult.`,
    rating: 6.7
  },
  {
    id: 5,
    user: {
      id: 5,
      name: `Paula Fleri-Soler`
    },
    dateTime: `2020-03-07T14:34:25.500Z`,
    quote: `It is certainly a magical and childlike way of storytelling, even if the content
              is a little more adult.`,
    rating: 8.9
  }
];
