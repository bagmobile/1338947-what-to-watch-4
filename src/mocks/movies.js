const mockReviews = [
  {
    author: `Kate Muir`,
    dateTime: `2020-03-07T14:34:25.500Z`,
    quote: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious
              Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in
              years.`,
    rating: 8.8
  },
  {
    author: `Bill Goodykoontz`,
    dateTime: `2020-03-07T14:34:25.500Z`,
    quote: `Anderson's films are too precious for some, but for those of us willing to lose
              ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added
              a hint of gravitas to the mix, improving the recipe.`,
    rating: 8.9
  },
  {
    author: `Matthew Lickona`,
    dateTime: `2020-03-07T14:34:25.500Z`,
    quote: `The mannered, madcap proceedings are often delightful, occasionally silly, and
              here and there, gruesome and/or heartbreaking.`,
    rating: 3.9
  },
  {
    author: `Amanda Greever`,
    dateTime: `2020-03-07T14:34:25.500Z`,
    quote: `I didn't find it amusing, and while I can appreciate the creativity, it's an
              hour and 40 minutes I wish I could take back.`,
    rating: 5
  },
  {
    author: `Paula Fleri-Soler`,
    dateTime: `2020-03-07T14:34:25.500Z`,
    quote: `It is certainly a magical and childlike way of storytelling, even if the content
              is a little more adult.`,
    rating: 6.7
  },
  {
    author: `Paula Fleri-Soler`,
    dateTime: `2020-03-07T14:34:25.500Z`,
    quote: `It is certainly a magical and childlike way of storytelling, even if the content
              is a little more adult.`,
    rating: 8.9
  }
];

const movies = [
  {
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Crime`,
    year: 2010,
    ratingScore: 4,
    ratingCount: 100,
    text: [
      `Text1.`,
      `Text2`
    ],
    director: `Director`,
    starring: [`Star 1`, `Star 2`, `Star 3`, `Star 4`, `Star 5`, `Star 6`, `Star 7`, `Star 8`],
    duration: `1h 25m`,
    reviews: mockReviews
  },
  {
    title: `Bohemian Rhapsody`,
    img: `img/bohemian-rhapsody.jpg`,
    poster: `img/bohemian-rhapsody.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Documentary`,
    year: 2011,
    ratingScore: 4.4,
    ratingCount: 200,
    text: [
      `Text1.`,
      `Text2`
    ],
    director: `Director`,
    starring: [`Star 1`, `Star 2`, `Star 3`, `Star 4`, `Star 5`, `Star 6`, `Star 7`, `Star 8`],
    duration: `1h 35m`,
    reviews: mockReviews
  },
  {
    title: `Macbeth`,
    img: `img/macbeth.jpg`,
    poster: `img/macbeth.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Crime`,
    year: 1987,
    ratingScore: 6.7,
    ratingCount: 300,
    text: [
      `Text1.`,
      `Text2`
    ],
    director: `Director`,
    starring: [`Star 1`, `Star 2`, `Star 3`, `Star 4`, `Star 5`, `Star 6`, `Star 7`, `Star 8`],
    duration: `1h 50m`,
    reviews: mockReviews
  },
  {
    title: `Aviator`,
    img: `img/aviator.jpg`,
    poster: `img/aviator.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Comedies`,
    year: 2014,
    ratingScore: 4.5,
    ratingCount: 5,
    text: [
      `Text1.`,
      `Text2`
    ],
    director: `Director`,
    starring: [`Star 1`, `Star 2`, `Star 3`, `Star 4`, `Star 5`, `Star 6`, `Star 7`, `Star 8`],
    duration: `1h 35m`,
    reviews: mockReviews
  },
  {
    title: `We need to talk about Kevin`,
    img: `img/we-need-to-talk-about-kevin.jpg`,
    poster: `img/we-need-to-talk-about-kevin.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Comedies`,
    year: 1990,
    ratingScore: 2,
    ratingCount: 34,
    text: [
      `Text1.`,
      `Text2`
    ],
    director: `Director`,
    starring: [`Star 1`, `Star 2`, `Star 3`, `Star 4`, `Star 5`, `Star 6`, `Star 7`, `Star 8`],
    duration: `1h 15m`,
    reviews: mockReviews
  },
  {
    title: `Johnny English`,
    img: `img/johnny-english.jpg`,
    poster: `img/johnny-english.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Sci-Fi`,
    year: 1981,
    ratingScore: 0,
    ratingCount: 54,
    text: [
      `Text1.`,
      `Text2`
    ],
    director: `Director`,
    starring: [`Star 1`, `Star 2`, `Star 3`, `Star 4`, `Star 5`, `Star 6`, `Star 7`, `Star 8`],
    duration: `45m`,
    reviews: mockReviews
  },
  {
    title: `What We Do in the Shadows`,
    img: `img/what-we-do-in-the-shadows.jpg`,
    poster: `img/what-we-do-in-the-shadows.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Romance`,
    year: 2011,
    ratingScore: 9,
    ratingCount: 2,
    text: [
      `Text1.`,
      `Text2`
    ],
    director: `Director`,
    starring: [`Star 1`, `Star 2`, `Star 3`, `Star 4`, `Star 5`, `Star 6`, `Star 7`, `Star 8`],
    duration: `1h 15m`,
    reviews: mockReviews
  },
  {
    title: `Revenant`,
    img: `img/revenant.jpg`,
    poster: `img/revenant.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Comedies`,
    year: 2017,
    ratingScore: 7.7,
    ratingCount: 3,
    text: [
      `Text1.`,
      `Text2`
    ],
    director: `Director`,
    starring: [`Star 1`, `Star 2`, `Star 3`, `Star 4`, `Star 5`, `Star 6`, `Star 7`, `Star 8`],
    duration: `2h 35m`,
    reviews: mockReviews
  },
  {
    title: `Movie`,
    img: `img/revenant.jpg`,
    poster: `img/revenant.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Dramas`,
    year: 2013,
    ratingScore: 7.7,
    ratingCount: 3,
    text: [
      `Text1.`,
      `Text2`
    ],
    director: `Director`,
    starring: [`Star 1`, `Star 2`, `Star 3`, `Star 4`, `Star 5`, `Star 6`, `Star 7`, `Star 8`],
    duration: `2h 35m`,
    reviews: mockReviews
  }
];

export const MAX_GENRES_LIST_SIZE = 9;

export const DEFAULT_GENRE = `All genres`;

export const genres = [...movies.reduce((acc, movie) => {
  acc.add(movie.genre);
  return acc;
}, new Set([DEFAULT_GENRE]))].slice(0, MAX_GENRES_LIST_SIZE);


export default movies;
