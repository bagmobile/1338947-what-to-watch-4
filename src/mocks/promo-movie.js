export const PromoMovie = {
  title: `The Grand Budapest Hotel`,
  img: `img/bg-the-grand-budapest-hotel.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  genre: `Crime`,
  year: 2010
};
