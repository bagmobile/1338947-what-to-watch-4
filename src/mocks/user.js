export const mockUserProfile = {"id": 1, "email": `test@test.ru`, "name": `test`, "avatar_url": `/wtw/static/avatar/5.jpg`};
