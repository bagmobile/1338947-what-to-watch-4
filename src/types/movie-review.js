import PropTypes from "prop-types";

const movieReviewShape = PropTypes.shape({
  id: PropTypes.number.isRequired,
  user: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired
  }),
  rating: PropTypes.number,
  dateTime: PropTypes.string.isRequired,
  quote: PropTypes.string.isRequired
});
export default movieReviewShape;
