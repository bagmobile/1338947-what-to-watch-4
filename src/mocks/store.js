import NameSpace from "../reducer/name-space";
import {mockMovies} from "./movies";
import {mockReviews} from "./movie-review";
import {DEFAULT_GENRE, DEFAULT_MOVIE_LIST_SIZE} from "../consts";
import {AuthorizationStatus} from "../reducer/user/user";

export const testStore = {
  [NameSpace.DATA]: {
    movies: mockMovies,
    promoMovie: mockMovies[0],
    reviews: mockReviews
  },
  [NameSpace.MOVIES_LIST]: {
    activeGenre: DEFAULT_GENRE,
    currentMovieListSize: DEFAULT_MOVIE_LIST_SIZE
  },
  [NameSpace.USER]: {
    authorizationStatus: AuthorizationStatus.NO_AUTH,
  }
};
