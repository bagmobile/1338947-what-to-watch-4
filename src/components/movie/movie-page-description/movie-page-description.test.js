import React from "react";
import renderer from "react-test-renderer";
import MoviePageDescription from "./movie-page-description";
import {mockMovies} from "../../../mocks/movies";
import {Tab} from "../../../consts";
import {mockReviews} from "../../../mocks/movie-review";

const movie = mockMovies[0];

describe(`MoviePageDescription component`, () => {

  it(`Render`, () => {
    const tree = renderer
      .create(<MoviePageDescription
        movie={movie}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render with custom activePage`, () => {
    const tree = renderer
      .create(<MoviePageDescription
        movie={movie}
        activePage={Tab.REVIEWS}
        reviews={mockReviews}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
