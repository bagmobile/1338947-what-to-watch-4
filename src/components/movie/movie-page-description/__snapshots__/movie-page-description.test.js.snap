// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePageDescription component Render 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item  movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item  false"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item  false"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      4.4
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Normal
      </span>
      <span
        className="movie-rating__count"
      >
        200 ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      description
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: Director
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: Star 1, Star 2, Star 3, Star 4, Star 5, Star 6, Star 7, Star 8
      </strong>
    </p>
  </div>,
]
`;

exports[`MoviePageDescription component Render with custom activePage 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item  false"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item  false"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item  movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-card__reviews movie-card__row"
  >
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Discerning travellers and Wes Anderson fans will luxuriate in the glorious
              Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in
              years.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Kate Muir
            </cite>
            <time
              className="review__date"
              dateTime="2020-03-07"
            >
              March 7, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8.8
        </div>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Anderson's films are too precious for some, but for those of us willing to lose
              ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added
              a hint of gravitas to the mix, improving the recipe.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Bill Goodykoontz
            </cite>
            <time
              className="review__date"
              dateTime="2020-03-07"
            >
              March 7, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8.9
        </div>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            The mannered, madcap proceedings are often delightful, occasionally silly, and
              here and there, gruesome and/or heartbreaking.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Matthew Lickona
            </cite>
            <time
              className="review__date"
              dateTime="2020-03-07"
            >
              March 7, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          3.9
        </div>
      </div>
    </div>
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            I didn't find it amusing, and while I can appreciate the creativity, it's an
              hour and 40 minutes I wish I could take back.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Amanda Greever
            </cite>
            <time
              className="review__date"
              dateTime="2020-03-07"
            >
              March 7, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          5
        </div>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            It is certainly a magical and childlike way of storytelling, even if the content
              is a little more adult.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Paula Fleri-Soler
            </cite>
            <time
              className="review__date"
              dateTime="2020-03-07"
            >
              March 7, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          6.7
        </div>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            It is certainly a magical and childlike way of storytelling, even if the content
              is a little more adult.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Paula Fleri-Soler
            </cite>
            <time
              className="review__date"
              dateTime="2020-03-07"
            >
              March 7, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8.9
        </div>
      </div>
    </div>
  </div>,
]
`;
